### PR TITLE
PLT-3782 Do not move the "@" sign and flag icon to the left of search bar when opening the right hand sidebar

### PR DESCRIPTION
**Pull request by khawerrind**
Now  "@" sign and flags icons stay on the right of the search box even when the right sidebar is opened

### CODE REVIEW

**Comment by yangchen1** [comment]
Hey @khawerrind, great work, thanks for your submission! I just noticed that when you open the @ or flagged posts in the RHS, the search bar seems to change size and have smaller padding around it (The search bar creeps closer to the @ symbol.) When you click on the @ or flagged posts, none of the elements should change size or positioning. Would you be able to fix that?

image

**Comment by khawerrind** [comment]
@yangchen1 sure checking the issue

**Comment by yangchen1** [comment]
Thanks @khawerrind!

**Comment by khawerrind** [comment]
@yangchen1 i have pushed the fix. The issue is that if the sidebar is not opened the search box is in the channel header and when the sider bar is opened the search box in the channel header gets hidden and the search box of the RHS becomes visible. So this was bit tricky to align exactly on the same points where the search of the channel header was. I tried to fixed the margin and padding issues using css. Have a look. Thanks

**Comment by yangchen1** [comment]
Ah I see - great fix @khawerrind :) Would it also be possible to make the channel members icon not shift to the left, either? If that means making the normal icon placement slightly to the left, I'm OK with that.

Before opening:
image

After opening:
image

Another thing I just noticed:
The app seems to break if you expand the sidebar, then click on the flag or @ again.

The current behaviour is to hide the flag / @ symbols (along with the rest of the channel header) on the expanded sidebar view. Doing that might be the easiest way to fix this, but ideally (because of [this ticket](https://github.com/mattermost/platform/issues/4363)) we would have them visible on the right side as you currently have it, but have it close the sidebar if you click on it in expanded view, instead of doing this:

image

**Comment by khawerrind** [comment]
@yangchen1 solved the remaining issues.

**Comment by yangchen1** [comment]
Hey @khawerrind, thank you so much for solving them. Looks super great, makes the UI so much more pleasant. Also, would you be able to get rid of the white space on the right side of the expanded view? (make the search bar longer?)

image

Finally, I noticed that when you close the expanded view via clicking @ or flag, and you re-open @ or flag, it'll open it _in_ the expanded view. I'm not sure if this should be the expected behaviour - I've queued it for our UX meeting tomorrow morning, and I'll circle back to you! Thank you so much for your patience and openness to feedback.

**Comment by khawerrind** [comment]
@yangchen1 let me know after the UI/UX meeting and i will solve both issues at once. Thanks

**Comment by yangchen1** [comment]
Hey @khawerrind, so we decided that the @ and flag should always open the RHS in the non-expanded view, so that'll have to be fixed. And the whitespace should be removed, as I mentioned before. Thanks!

**Comment by khawerrind** [comment]
@yangchen1 ok working on another issue after that will fix this

**Comment by yangchen1** [comment]
@khawerrind Sounds great! :)

**Comment by khawerrind** [comment]
@yangchen1 fixed all the remaining issues. Please check. Thanks!

**Comment by yangchen1** [comment]
+1. @khawerrind Amazing work, something I was looking forward to for months. Beautiful.

**Comment by khawerrind** [comment]
@yangchen1 Thank you! glad to have helped.

File: webapp/components/search_bar.jsx [hunk 1]
```diff
@@ -208,7 +300,18 @@ export default class SearchBar extends React.Component {
...
         );
     }
 }
+
+SearchBar.defaultProps = {
+    showMentionFlagBtns: true
+};
+
+SearchBar.propTypes = {
+    showMentionFlagBtns: React.PropTypes.func
```
Comment (jwilander) [inline webapp/components/search_bar.jsx]: Should this be `React.PropTypes.bool` instead?

Comment (khawerrind) [inline webapp/components/search_bar.jsx]: @jwilander sorry for this. I overlooked that one. Fixed now.

**Comment by jwilander** [comment]
Thanks, looks good!

**Comment by enahum** [comment]
awesome thanks @khawerrind

**Pull request merged**